### PR TITLE
added course context and file structure

### PR DESCRIPTION
**Pull request by SanaBob**
- changed file structure to include course and courseId folder - task 1
- added useCourse hook to fetch course context - task 2

### CODE REVIEW

File: yarn.lock
Comment (Opeyem1a) [inline yarn.lock:1]: why did your yarn lock change when your package.json didnt. If no explanation, revert this and rerun yarn install locally

File: src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import { useParams } from "next/navigation"
+import { PropsWithChildren, createContext, useContext, useEffect, useState } from "react"
+
+type CourseContextType = {
+    courseId: number
+    courseName: string
+}
+
+const CourseContext = createContext<CourseContextType>({
+  courseId: 0,
```
Comment (Opeyem1a) [inline src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx:12]: default this to null and just don't send the request to fetch the course while courseId is null

File: src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+})
+
+const useCourseProvider = (): CourseContextType => {
+  const { courseId } = useParams<{ courseId: string }>()
+  const [courseName, setCourseName] = useState<string>('')
+
+  useEffect(() => {
+    const fetchCourse = async () => {
+      try {
+        const courseResponse = await fetch(`${process.env.NEXT_PUBLIC_BACKEND_URL}/api/v1/courses/?id=${courseId}`,)
```
Comment (Opeyem1a) [inline src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx:23]: wait I feel like you're calling this api endpoint wrong.
I thought django did `.../courses/1` by default, we shouldn't be using query params for this

File: src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        const courseData = await courseResponse.json()
+        setCourseName(courseData[0].name)
```
Comment (Opeyem1a) [inline src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx:25]: If the previous comment is done, the response type should not be a list, either

File: src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      } catch (e) {
+        console.error(e)
+      }
+    }
+    fetchCourse()
+  }, [courseId])
```
Comment (Opeyem1a) [inline src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx:31]: also depends on `setCourseName`

File: src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+  return {
+    courseId: parseInt(courseId),
```
Comment (Opeyem1a) [inline src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx:34]: what happens when the user enters an invalid integer as the course id?

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

File: src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        console.error(e)
+      }
+    }
+    if (courseId && !isNaN(Number(courseId))) {
+      fetchCourse()
+    }
+  }, [courseId, setCourseName])
+
+  return {
+    courseId: !isNaN(Number(courseId)) ? Number(courseId) : null,
```
Comment (Opeyem1a) [inline src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx:36]: nit;

consider that you're doing this validation only partially. A valid course id would be a positive integer specifically, and this just validated that it's a number at all

**Review by Opeyem1a — APPROVED** [review]

File: src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx (hunk 6 above)
Comment (SanaBob) [reply src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx:36]: I'll address this in the next pr

**Pull request merged**